### PR TITLE
Reset backoff when connectivity status changes

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
Port of https://github.com/firebase/firebase-ios-sdk/pull/5857

### CODE REVIEW

**Comment by schmidt-sebastian** [comment]
Verified via Fireeats.

File: firebase-firestore/CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,7 @@
 # Unreleased
+- [fixed] Removed a delay that may have prevented Firestore from immediately
+  establishing a network connection if a connectivity change occurred while
```
Comment (var-const) [inline firebase-firestore/CHANGELOG.md:3]: Nit: perhaps s/establishing/reestablishing/?

Comment (schmidt-sebastian) [reply firebase-firestore/CHANGELOG.md]: Done

**Comment by schmidt-sebastian** [comment]
Merging as this seems to hit a build problem in the current tests.

**Pull request merged**